version: "3.8"

services:
  traefik:
    image: traefik:v3.4
    restart: always

    command:
      # Docker provider
      - '--providers.docker=true'
      - '--providers.docker.exposedbydefault=false'

      # Entrypoints
      - '--entrypoints.web.address=:80'
      - '--entrypoints.websecure.address=:443'

      # ACME HTTP Challenge
      - '--certificatesresolvers.http_resolver.acme.email=${ACME_EMAIL}'
      - '--certificatesresolvers.http_resolver.acme.storage=/acme/acme.json'
      - '--certificatesresolvers.http_resolver.acme.httpchallenge=true'
      - '--certificatesresolvers.http_resolver.acme.httpchallenge.entrypoint=web'

      # Dashboard
      - '--api.dashboard=false'
      - '--api.insecure=false' # pas exposé en HTTP non sécurisé

      # Logs
      - '--log.level=INFO'
    ports:
      - '80:80'
      - '443:443'

    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./acme:/acme

    environment:
      - ACME_EMAIL=${ACME_EMAIL}
      - DOMAIN_NAME=${DOMAIN_NAME}

    networks:
      - public

  app:
    build:
      context: .
      dockerfile: Dockerfile
    restart: unless-stopped
    ports:
      - 3000
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.api.rule=Host(`${DOMAIN_NAME}`)'
      - 'traefik.http.routers.api.entrypoints=web,websecure'
      - 'traefik.http.routers.api.tls=true'
      - 'traefik.http.routers.api.tls.certresolver=http_resolver'
      # - "traefik.http.services.api.loadbalancer.server.port=8080"
    networks:
      - public

networks:
  public:
    driver: bridge
    attachable: true
